### PR TITLE
Animation support

### PR DESCRIPTION
**Pull request by HurrellT**
Adding support for css keyframes for animations

### CODE REVIEW

File: source/RenoirSt/CssSelector.class.st [hunk 1]
```diff
@@ -169,6 +169,12 @@ CssSelector >> focus [
 	^ CssPseudoClassSelector focusOn: self
 ]
 
+{ #category : #'as yet unclassified' }
```
Comment (gcotelli) [inline source/RenoirSt/CssSelector.class.st]: This method is missing the category

File: source/RenoirSt/CssSelector.class.st [hunk 2]
```diff
@@ -259,6 +265,12 @@ CssSelector >> target [
 	^ CssPseudoClassSelector targetOn: self
 ]
 
+{ #category : #'as yet unclassified' }
```
Comment (gcotelli) [inline source/RenoirSt/CssSelector.class.st]: This method is missing the category

File: source/RenoirSt/CssKeyframesRule.class.st [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	^ self new
+		initializeNamed: aName
+		enabling: aCascadingStyleSheet
+		withWebkitCompatibilityEnabled: aBoolean
+]
+
+{ #category : #Printing }
+CssKeyframesRule >> appropriateSelector [
+
+	^webkitCompatibility ifTrue: [ '@-webkit-keyframes' ] ifFalse: [ '@keyframes' ]
```
Comment (gcotelli) [inline source/RenoirSt/CssKeyframesRule.class.st]: Do we really need this web-kit-compatibility flag? I'm in favor of always use the standard option (`@keyframes`) and to enable some kind of vendor mode if needed that adds the webkit  and mozilla rules. But I really hope that keyframes work on any modern browser. What do you think?

**Review by gcotelli — CHANGES_REQUESTED** [review]
I saw some examples passing 'rotate(270deg)' as a string. If there's a list of such functions we can add support for them (we already have something similar for math functions). I would love to write it like `CssRotate by: 207 deg`

File: source/RenoirSt/CssSelector.class.st (hunk 2 above)
Comment (gcotelli) [reply source/RenoirSt/CssSelector.class.st]: It's also missing a test case

File: source/RenoirSt/CssSelector.class.st (hunk 1 above)
Comment (gcotelli) [reply source/RenoirSt/CssSelector.class.st]: It's also missing a test case

**Comment by gcotelli** [comment]
Another thing I will change is that the keyframe selector must be one of: `from` `to` or a percentage. The current implementation provides the standard CssSelector builder so you can combine anything in invalid ways and write invalid CSS.

File: source/RenoirSt/CssKeyframesRule.class.st (hunk 3 above)
Comment (HurrellT) [reply source/RenoirSt/CssKeyframesRule.class.st]: Flag removed in https://github.com/ba-st/RenoirSt/pull/34/commits/6fa379f18ffab27cacbc4db41f8c49b5baa55081

File: source/RenoirSt/CssSelector.class.st (hunk 2 above)
Comment (HurrellT) [reply source/RenoirSt/CssSelector.class.st]: Fixed in https://github.com/ba-st/RenoirSt/pull/34/commits/6fa379f18ffab27cacbc4db41f8c49b5baa55081

File: source/RenoirSt/CssSelector.class.st (hunk 1 above)
Comment (HurrellT) [reply source/RenoirSt/CssSelector.class.st]: Fixed in https://github.com/ba-st/RenoirSt/pull/34/commits/6fa379f18ffab27cacbc4db41f8c49b5baa55081

**Review by gcotelli — APPROVED** [review]

**Pull request merged**